pkg/server/bootstrap/identity.go

```diff
@@ -282,7 +282,11 @@ func decorateWildcardPathsWithResourceIdentities(urlPath string, ids *identities
 		if len(id) == 0 {
 			return "", fmt.Errorf("identity for %s is unknown", gr)
 		}
-		comps[5] += ":" + id
+
+		// Check if identity already exists in the URL path before appending it.
+		if !strings.Contains(comps[5], ":") {
```

Comment:
Can you elaborate on this? I think its not user passed one but system added one here https://github.com/kcp-dev/kcp/blob/bc50273a7a1fe2f192ccf4bb4c00120d45607e7d/pkg/virtual/framework/forwardingregistry/store.go#L279
So at this place we trying to append it second time

Reply:
This is an API exposed to the world, right? Any client can hit this code, not just those sent by the service.

Reply:
Took me a while but I got what you had in mind. 
TBH I think we should leave users identity, as it might drive systems behaviour to undetermenistic possition - "I have bad identity, but it still works somehow".  So if users provided something wrong, let it fail, as it is clear what happening. 

Happy to change, but I don't this "swapping bad identity with working one" will help users to understand API beahviour.